collated/main/limcel.md

```diff
@@ -0,0 +1,928 @@
+    }
+
+    //====================================== HELPER METHODS =====================================
+    /**
+     * Converts the schedule list output by replacing all occurrence of "," with ": " for better UI visualisation.
+     */
+    public static String changeToAppropriateUiFormat(String value) {
+        value = value.replace(",", "\n");
+        return value;
+    }
+}
+```
+###### \java\seedu\address\logic\Logic.java
+``` java
+    /** Returns an unmodifiable view of the schedule list */
+    ObservableList<Schedule> getScheduleList();
+```
+###### \java\seedu\address\logic\LogicManager.java
+``` java
+    @Override
+    public ObservableList<Schedule> getScheduleList() {
+        return model.getScheduleList(); }
+```
+###### \java\seedu\address\logic\parser\ScheduleCommandParser.java
+``` java
+/**
+ * Parses input arguments and creates a new ScheduleCommand object
+ */
+public class ScheduleCommandParser implements Parser<ScheduleCommand> {
+
+    public static final DateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+    private List<DateGroup> scheduleToParse;
+
+    /**t
+     * Parses the given {@code String} of arguments in the context of the ScheduleCommand
+     * and returns an ScheduleCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public ScheduleCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_SCHEDULE);
+        Index index;
+
+        try {
+            args = args.trim();
+            String[] str = args.split(" ");
+            // string length should be at least of length three comprising of command, index and scheduled date,time:
+            // e.g. Schedule, 1, d/28October2019 3pm
+            if (str.length == 0 && str.length < 3) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, ScheduleCommand.MESSAGE_USAGE));
+            // it is a valid Command
+            } else if (str.length >= 3 && isNumeric(str[0])) {
+                index = ParserUtil.parseIndex(argMultimap.getPreamble());
+                com.joestelmach.natty.Parser parser = new com.joestelmach.natty.Parser();
+                scheduleToParse = parser.parse(argMultimap.getValue(PREFIX_SCHEDULE).get());
+                // Natty returns no date or returns with more than one date --> invalid
+                if (scheduleToParse.isEmpty() || scheduleToParse.size() > 1) {
+                    throw new ParseException("Please enter a more specific date for your student's consultation.");
+                }
+                Calendar date = Calendar.getInstance();
+                date.setTime(scheduleToParse.get(0).getDates().get(0));
+                return new ScheduleCommand(index, date);
```

Comment:
This method is quite long, perhaps you can shorten it by abstracting out this code chunk (line 238 - 247) into a method?